hydra_agent/requests.py

```diff
@@ -0,0 +1,100 @@
+import urllib.request
+import json
+import logging
+from urllib.error import URLError, HTTPError
+from redis_proxy import RedisProxy
+from graphutils import GraphUtils
+from redisgraph import Graph, Node
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class Requests:
+
```

Comment:
we don't need to reimplement this from scratch, use the library [`requests` 3](https://3.python-requests.org/).
Add `requests` to the requirements, then you just need:
```
from requests import Session

class Agent(Session):
    ....
```
This way you will have all the verbs and session management already implemented as [defined here](http://3.python-requests.org/api/#request-sessions).

First rule: we don't reinvent the wheel, always look for an existing popular and maintained library.

Reply:
Yes, agree. I was planning to do something like this:

requests.py:

```
from requests import Session

class Requests(Session):
    def get():

```


agent.py:
```
from hydra_agent.requests import Requests

class HydraAgent(Requests):
    ....

```

hydra_agent.requests will inherit requests3 session and already encapsulate basic Redis Interactions. 

agent will inherit hydra_agent.requests and have everything that session supports with basic Redis interactions already embedded in GET, POST and so on.. Do you agree?

Reply:
for bringing Redis in the picture you can use multiple inheritance or "mixin":
```
class Agent(Session, RedisProxy):
     ...
```
be careful about precedence in Python classes.

Reply:
https://stackoverflow.com/questions/533631/what-is-a-mixin-and-why-are-they-useful

Reply:
in particular our use-case is listed in this comment https://stackoverflow.com/questions/533631/what-is-a-mixin-and-why-are-they-useful#comment27661661_547714

Reply:
Okay, I'm studying the options and will experiment